tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/TagMapTest.java

```diff
@@ -53,12 +53,22 @@ void testSingleton() {
 
     @Test
     void testExtraEntryKeyAlreadyExists() {
-        TagMap map = TagMap.of(Collections.singletonMap("foo", "bar"));
-        assertThatThrownBy(() -> map.withEntry("foo", "baz")).isInstanceOf(IllegalArgumentException.class);
+        TagMap original = TagMap.of(Collections.singletonMap("foo", "bar"));
+        TagMap updated = original.withEntry("foo", "baz");
+        assertThat(updated)
+                .isEqualTo(ImmutableMap.of("foo", "baz"))
+                .as("Original must not be mutated")
+                .isNotSameAs(original);
```

Comment:
Is it worth testing state of original is not mutated? Thinking something like the following:

```suggestion
        assertThat(updated)
                .hasSize(1)
                .containsEntry("foo", "baz")
                .isEqualTo(ImmutableMap.of("foo", "baz"))
                .as("Original must not be mutated")
                .isNotSameAs(original)
                .isNotEqualTo(original);
        assertThat(original)
                .hasSize(1)
                .as("Original must not be mutated")
                .containsEntry("foo", "bar")
                .isNotEqualTo(updated);
```

Reply:
Good idea, thanks!